tasks/testwasher.py

```diff
@@ -0,0 +1,86 @@
+from collections import defaultdict
+import json
+import yaml
+from typing import List
+from tasks.test_core import ModuleTestResult
+
+
+class TestWasher:
+    def __init__(
+        self,
+        test_output_json_file="module_test_output.json",
+        flaky_test_indicator="flakytest: this is a known flaky test",
+        flakes_file_path="flakes.yaml",
+    ):
+        self.test_output_json_file = test_output_json_file
+        self.flaky_test_indicator = flaky_test_indicator
+        self.flakes_file_path = flakes_file_path
+        self.known_flaky_tests = defaultdict(set)
+
+        self.parse_flaky_file()
+
+    def get_non_flaky_failing_tests(self, module_path):
+        """
+        Parse the test output json file and compute the failing tests and the one known flaky
+        """
+        failing_tests = defaultdict(set)
+        flaky_marked_tests = defaultdict(set)
+        with open(f"{module_path}/{self.test_output_json_file}", encoding='utf-8') as f:
+            for line in f:
+                test_result = json.loads(line)
+                # TODO: tests can be retried we should handle that
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** This will be handled in another PR ?‏

Reply:
Good point, I can handle it in this PR. Was a reminder for me to not forget it and I did forget it 😓